### PR TITLE
Efischer/watcher timeout

### PR DESCRIPTION
**Pull request by efischer19**
WIP, nomerge pls

### CODE REVIEW

**Comment by efischer19** [comment]
This is seemingly working for me locally, when paired with https://github.com/edx/edx-platform/pull/16051

File: Makefile [hunk 1]
```diff
@@ -119,12 +122,18 @@ update-db: | studio-update-db lms-update-db discovery-update-db ecommerce-update
 lms-shell: ## Run a shell on the LMS container
 	docker exec -it edx.devstack.lms env TERM=$(TERM) /edx/app/edxapp/devstack.sh open
 
+lms-watcher-shell: ## Run a shell on the LMS watcher container
+	docker exec -it edx.devstack.lms_watcher env TERM=$(TERM) /edx/app/edxapp/devstack.sh open
+
 lms-attach: ## Attach to the LMS container process to use the debugger & see logs.
-	docker attach `docker ps -aqf "name=edx.devstack.lms"`
+	docker attach `docker ps -aqf "name=edx.devstack.lms$$"`
```
Comment (gsong) [inline Makefile]: Why not just attach directly to the container name: `docker attach edx.devstack.lms`?

Comment (efischer19) [reply Makefile]: That's a good question! @doctoryes, I think you added this logic in https://github.com/edx/devstack/pull/198 - does George's suggestion seem reasonable, or was there a specific reason for filtering the results of `docker ps`?

Comment (doctoryes) [reply Makefile]: No specific reason - guess I didn't know that the container name itself would just work.

Comment (doctoryes) [reply Makefile]: Simpler is nearly always better!

**Comment by macdiesel** [comment]
@efischer19 do you think there should be a target to stop just the watchers? dev.watchers.stop or something similar.

**Comment by macdiesel** [comment]
Or a target to start just the watchers? I guess you would just run dev.up.all even if you had already had the other set of containers already running?

**Comment by efischer19** [comment]
Yeah, that's the part I'm least sure of - what's most useful to most workflows? Should we add both the `dev.up.all` target as well as a new watchers-only target? Just the latter? How have we been building the Makefile to this point?

I'll update so that dev.up.all/stop both work for *everything*, and give the watchers their own start/stop as well. Totally open to further discussion here, but IMO it feels like it'll be less hassle to have too many targets than it would be to not have enough.

**Comment by gsong** [comment]
I like @macdiesel's idea. That way `dev.up.all` can just be a compositional task.

**Comment by efischer19** [comment]
Updated. `dev.up` and `stop` both continue to function as before. New targets `dev.up.watchers` and `stop.watchers` allow you to start and stop asset watching as desired, while `dev.up.all` and `stop.all` combine the previous targets compositionally.

**Review by macdiesel — APPROVED** [review]
Thanks for the PR @efischer19.

File: Makefile [hunk 2]
```diff
@@ -54,11 +59,16 @@ provision: | dev.provision ## This command will be deprecated in a future releas
 
 stop: ## Stop all services
 	(test -d .docker-sync && docker-sync stop) || true ## Ignore failure here
-	docker-compose stop
+	docker-compose -f docker-compose.yml stop
```
Comment (gsong) [inline Makefile:62]: Remove `-f` option.

File: Makefile [hunk 3]
```diff
@@ -38,6 +38,11 @@ dev.repo.reset: ## Attempts to reset the local repo checkouts to the master work
 dev.up: | check-memory ## Bring up all services with host volumes
 	docker-compose -f docker-compose.yml -f docker-compose-host.yml up -d
 
+dev.up.watchers: | check-memory ## Bring up asset watcher containers
+	docker-compose docker-compose-watchers.yml up -d
```
Comment (gsong) [inline Makefile:42]: Is this missing a `-f`?

**Pull request merged**